doc/src/index.md

```diff
@@ -72,6 +72,7 @@
   * Documentation of Julia's Internals
       * [Initialization of the Julia runtime](@ref)
       * [Eval of Julia code](@ref)
+      * [High-level Overview of the Native-Code Generation Process](@ref)
```

Comment:
I might put this after the memory layout section, since that and the preceding sections are helpful to understand prior to reading this (or at least it seems that way to me)